composer.json

```diff
@@ -42,6 +42,8 @@
             "src/"
         ]
     },
+    "minimum-stability": "dev",
```

Comment:
Allow me rewrite your question

Why both settings are needed?

It download the sources when is not possible resolve an stable version (when circular dependencies occur)